lib/rust/flexer/src/group.rs

```diff
@@ -34,70 +178,40 @@ pub mod rule;
 #[derive(Clone,Debug,Default)]
 pub struct Group {
     /// A unique identifier for the group.
-    pub id: usize,
+    pub id: Identifier,
     /// A name for the group (useful in debugging).
     pub name: String,
     /// The parent group from which rules are inherited.
-    pub parent: Option<Box<Group>>,
+    ///
+    /// It is ensured that the group is held mutably.
+    pub parent_index: Option<usize>,
     /// A set of flexer rules.
     pub rules: Vec<Rule>,
 }
 
 impl Group {
 
+    /// Creates a new group.
+    pub fn new(id:Identifier, name:String, parent_index:Option<usize>) -> Self {
```

Comment:
this API requires user to provide here `String`, so you have ugly code like `Group::new(x,"foo".into(),None)`. Instead, you should use `name:impl Into<String>` here, which would allow passing both `String` and `&str` here.